packages/app/src/background/services/credentials/index.ts

```diff
@@ -62,197 +68,216 @@ export default class VerifiableCredentialsService implements IBackupable {
 
-    if (!cryptkeeperVerifiableCredentials.has(verifiableCredentialHash)) {
+    if (!cryptkeeperVCs.has(vcHash)) {
       throw new Error("Verifiable Credential does not exist.");
     }
 
-    cryptkeeperVerifiableCredentials.delete(verifiableCredentialHash);
-    await this.writeCryptkeeperVerifiableCredentials(cryptkeeperVerifiableCredentials);
-    await this.historyService.trackOperation(OperationType.DELETE_VERIFIABLE_CREDENTIAL, {});
-    await this.notificationService.create({
-      options: {
-        title: "Verifiable Credential deleted.",
-        message: `Deleted 1 Verifiable Credential.`,
-        iconUrl: browser.runtime.getURL("/icons/logo.png"),
-        type: "basic",
-      },
+    cryptkeeperVCs.delete(vcHash);
+    await this.writeCryptkeeperVCs(cryptkeeperVCs);
+
+    await this.announce({
+      notificationTitle: "Verifiable Credential deleted.",
+      notificationMessage: `Deleted 1 Verifiable Credential.`,
+      notificationType: "basic",
+      historyOperationType: OperationType.DELETE_VERIFIABLE_CREDENTIAL,
     });
   };
 
-  deleteAllVerifiableCredentials = async (): Promise<void> => {
-    const cryptkeeperVerifiableCredentials = await this.getCryptkeeperVerifiableCredentialsFromStore();
+  /**
+   * Deletes all Verifiable Credentials from the store.
+   * @returns A promise that resolves when all Verifiable Credentials are deleted.
+   * @throws An error if there are no Verifiable Credentials to delete.
+   */
+  deleteAllVCs = async (): Promise<void> => {
+    const cryptkeeperVCs = await this.getCryptkeeperVCsFromStore();
 
-    if (cryptkeeperVerifiableCredentials.size === 0) {
+    if (cryptkeeperVCs.size === 0) {
       throw new Error("No Verifiable Credentials to delete.");
     }
 
     await this.verifiableCredentialsStore.clear();
-    await this.historyService.trackOperation(OperationType.DELETE_ALL_VERIFIABLE_CREDENTIALS, {});
-    await this.notificationService.create({
-      options: {
-        title: "All Verifiable Credentials deleted.",
-        message: `Deleted ${cryptkeeperVerifiableCredentials.size} Verifiable Credential(s).`,
-        iconUrl: browser.runtime.getURL("/icons/logo.png"),
-        type: "basic",
-      },
+    await this.announce({
+      notificationTitle: "All Verifiable Credentials deleted.",
+      notificationMessage: `Deleted ${cryptkeeperVCs.size} Verifiable Credential(s).`,
+      notificationType: "basic",
+      historyOperationType: OperationType.DELETE_ALL_VERIFIABLE_CREDENTIALS,
     });
   };
 
-  generateVerifiablePresentationRequest = async ({ request }: IVerifiablePresentationRequest): Promise<void> => {
+  /**
+   * Handles a request for a Verifiable Presentation. Opens a popup to confirm the request.
+   * @param request - The string representing a request for a Verifiable Presentation.
+   * @returns A promise that resolves when the popup is opened.
+   */
+  handleVPRequest = async ({ request }: IVerifiablePresentationRequest): Promise<void> => {
```

Comment:
Why this service contains VPs with VCs? It should be a separate service.

Reply:
I think conceptually they belong in the same service since VPs are just a data field with the VC definition (see the data model here: https://www.w3.org/TR/vc-data-model/#presentations-0). I think together they are both doing "one thing", that is interfacing with the Verifiable Credential standard.